Palindrome.java

```diff
@@ -0,0 +1,132 @@
+package WEEK1;
+
+import static org.testng.Assert.assertEquals;
+
+//import java.util.Scanner;
+
+import org.junit.Test;
+
+public class Palindrome {
+	static int length = 0;
+	
+	public static boolean is_palindrome(Node first, Node second){
+		Node temp_f = first;
+		Node temp_s = second;
+		
+		while(temp_f!=null && temp_f.next!=second){
+			if(temp_f.data == temp_s.data){
+				temp_f = temp_f.next;
+				temp_s = temp_s.next;
+			}
+			else{
+				return false;
+			}
+		}
+		return true;
+	}
+	
+	public static Node reverse(Node start){
+		Node current = start;
+		Node prev = null;
+		Node next;
+		while(current!=null){
+			next = current.next;
+			current.next = prev;
+			prev = current;
+			current = next;
+		}
+		
+		start = prev;
+		return start;
+	}
+	
+	public static boolean palindrome(Node start){
+		if(start==null){
+			return true;
+		}
+		if(start.next==null){
+			return false;
+		}
+		Node last = start;
+		Node temp = start;
+		int n = (length/2);
+		int l = 0;
+		while(l<n){
+			l++;
```

Comment:
if you know how much you are iterating (0<=i<n), you should be using a for loop here.